### PR TITLE
operator: moved CRD to subchart

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
- moved operator CRD to subchart
- replaced flag `crd.enabled` to `crds.plain`, which instead of disabling CRDs templating, changes a mode in which CRDs should be rendered: as plain text CRDs or as templates

### CODE REVIEW

**Review by zekker6 — COMMENTED** [review]
Could you describe some benefits or issues which new type of CRDs deployment solves?
I'm not sure if I completely understand what exactly this PR addresses.

File: charts/victoria-metrics-operator/values.yaml [hunk 1]
```diff
@@ -28,9 +28,11 @@ image:
...
   pullPolicy: IfNotPresent
 
-crd:
-  # -- Enables CRD creation and management. With this option, if you remove this chart, all CRD resources will be deleted with it.
-  create: true
+crds:
+  # -- check if plain or templated CRDs should be created.
+  # with this option set to `false`, all CRDs will be rendered from templates.
+  # with this option set to `true`, all CRDs are immutable and require manual upgrade.
+  plain: false
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-operator/values.yaml:35]: yes, next step is CRD management removal from k8s-stack

**Comment by AndrewChubatiuk** [comment]
> Could you describe some benefits?

single place, with CRDs instead of having them both in operator and k8s stack managed differently (as templates in operator and as plain CRDs in k8s-stack)

File: charts/victoria-metrics-operator/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Haleygo) [reply charts/victoria-metrics-operator/values.yaml:35]: >  check if plain or templated CRDs should be created.

I don't think user can understand the `crds.plain` based on the name and description. 
And from the operator chart's perspective, setting `crds.plain=true` offers no advantages over `crds.plain=false`, so it's only for k8s-stack chart to use, right? In that case, I still don't see the necessity to combine the two crd sources.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**